src/instr/instr_private.h

```diff
@@ -262,6 +262,14 @@ class NewEvent : public PajeEvent  {
 
 };
 
+class paje_value{
```

Comment:
Seriously?? Is this code object oriented in any way? Where are the constructors, the instances and the destructor?

Please make val_t a class, change PJ_value_new into the constructor of that class, and the two other ones into static methods. And while you're at it, make every fields private, and add getter methods.